libsession/src/main/java/org/session/libsession/messaging/sending_receiving/notifications/PushRegistryV1.kt

```diff
@@ -0,0 +1,147 @@
+package org.session.libsession.messaging.sending_receiving.notifications
+
+import android.annotation.SuppressLint
+import nl.komponents.kovenant.Promise
+import okhttp3.MediaType
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.session.libsession.messaging.MessagingModuleConfiguration
+import org.session.libsession.snode.OnionRequestAPI
+import org.session.libsession.snode.OnionResponse
+import org.session.libsession.snode.Version
+import org.session.libsession.utilities.Device
+import org.session.libsession.utilities.TextSecurePreferences
+import org.session.libsignal.utilities.JsonUtil
+import org.session.libsignal.utilities.Log
+import org.session.libsignal.utilities.emptyPromise
+import org.session.libsignal.utilities.retryIfNeeded
+import org.session.libsignal.utilities.sideEffect
+
+@SuppressLint("StaticFieldLeak")
+object PushRegistryV1 {
+    private val TAG = PushRegistryV1::class.java.name
+
+    val context = MessagingModuleConfiguration.shared.context
+    private const val maxRetryCount = 4
+
+    private val server = Server.LEGACY
+
+    fun register(
+        device: Device,
+        isUsingFCM: Boolean = TextSecurePreferences.isPushEnabled(context),
+        token: String? = TextSecurePreferences.getPushToken(context),
+        publicKey: String? = TextSecurePreferences.getLocalNumber(context),
+        legacyGroupPublicKeys: Collection<String> = MessagingModuleConfiguration.shared.storage.getAllClosedGroupPublicKeys()
+    ): Promise<*, Exception> = when {
+        isUsingFCM -> retryIfNeeded(maxRetryCount) {
+            android.util.Log.d(
```

Comment:
We can probably get rid of this one when it's working, or at least reduce the amount of data we log. Also we should switch to the libsignal `Log` instead of the `android.util.Log` one